tests/Integration/Behaviour/Features/Scenario/CatalogPriceRule/catalog_price_rules_listing.feature

```diff
@@ -0,0 +1,307 @@
+      | reduction type            | amount               |
+      | reduction value           | 50                   |
+      | shop                      | testShop             |
+      | includes tax              | true                 |
+      | price                     | 10                   |
+    When I add catalog price rule "catalogPriceRuleReference3" with following details:
+      | name            | catalog price rule 3  |
+      | currency        | usd                   |
+      | country         | UnitedStates          |
+      | group           | visitor               |
+      | from quantity   | 5                     |
+      | reduction type  | amount                |
+      | reduction value | 10                    |
+      | shop            | testShop              |
+      | includes tax    | true                  |
+      | price           | 20                    |
+    And I add catalog price rule "catalogPriceRuleReference4" with following details:
+      | name            | catalog price rule 4  |
+      | currency        | usd                   |
+      | country         | UnitedStates          |
+      | group           | visitor               |
+      | from quantity   | 5                     |
+      | reduction type  | amount                |
+      | reduction value | 10                    |
+      | shop            | testShop              |
+      | includes tax    | true                  |
+      | price           | 20                    |
+    And I add catalog price rule "catalogPriceRuleReference5" with following details:
+      | name            | catalog price rule 5  |
+      | currency        | usd                   |
+      | country         | UnitedStates          |
+      | group           | visitor               |
+      | from quantity   | 5                     |
+      | reduction type  | amount                |
+      | reduction value | 10                    |
+      | shop            | testShop              |
+      | includes tax    | true                  |
+      | price           | 20                    |
+    Then I should be able to see following list of catalog price rules with language "en" with limit 2 offset 0 and total 5 and product "product1":
+      | name                  | currency   | country       | group    | from quantity  | reduction type| reduction value | shop     | includes tax| price | from                | to                  |
+      | catalog price rule 1  | usd        | UnitedStates  | visitor  | 1              | amount        | 111.50          | testShop | true        | 50    | 0000-00-00 00:00:00 | 0000-00-00 00:00:00 |
+      | catalog price rule 2  | usd        | UnitedStates  | visitor  | 1              | amount        | 50              | testShop | true        | 10    | 0000-00-00 00:00:00 | 0000-00-00 00:00:00 |
+    And I should be able to see following list of catalog price rules with language "en" with limit 50 offset 0 and total 5 and product "product1":
```

Comment:
I think you should add the same test for `product2` just to ensure that it has the same rules initially before you start adding some conditions that reduce its list

Reply:
Good point!

Reply:
I think only this point is left to be handled, check all the catalog price rules for product2 (with limit 50 and offset 0) to check they match the ones product1 has as well